client/flutter/lib/pages/onboarding/notifications_page.dart

```diff
@@ -35,4 +54,13 @@ class _NotificationsPageState extends State<NotificationsPage> {
   void _complete() {
     widget.onNext();
   }
+
+  void _registerFCMToken() async {
+    await _firebaseMessaging.getToken().then((String token) {
+      assert(token != null);
```

Comment:
will this throw an Exception that isn't handled if the user does not accept the notification permission?

Reply:
No, they happen in parallel. In the worst case (this fires but user opts out - now or later - push message is sent by Apple but never seen by the user. Behind the scenes Firebase will get a notification that he device ID has opted out. Firebase is supposed to remove it from sending in the future. If Firebase keeps sending user never sees it, regardless.

Reply:
Sorry, still familiarizing myself with Dart, and just read [in the docs](https://dart.dev/guides/language/language-tour#assert) that assert statements are only evaluated in development:

> In production code, assertions are ignored, and the arguments to assert aren’t evaluated.

Will the `WhoService.putDeviceToken(token);` handle `null` tokens? And/or, is there a scenario where the token would be `null`?